### PR TITLE
Show additional data nodes in DAG

### LINKED ISSUE #787: Scenario DAG - add additional_datanodes in DAG
**Description**

Additional data nodes should be added to the scenario DAG.

image

```python
from taipy.config import Config
import taipy as tp
from taipy.gui import Gui, notify

#Config.configure_job_executions(mode="standalone")


# Normal function used by Taipy
def double(nb):
    return nb * 2


# Configuration of Data Nodes
input_cfg = Config.configure_data_node(id="somedata", default_data=21)
output_cfg = Config.configure_data_node(id="result")


# Configuration of tasks
first_task_cfg = Config.configure_task(id="double",
                                       function=double,
                                       input=input_cfg,
                                       output=output_cfg)


other_cfg = Config.configure_data_node(id="other")

# Configuration of scenario
scenario_cfg = Config.configure_scenario(id="my_scenario",
                                         task_configs=[first_task_cfg],
                                         additional_data_node_configs=[other_cfg],
                                         name="my_scenario")




if __name__=="__main__":
    tp.Core().run()
    scenario_1 = tp.create_scenario(scenario_cfg)

    scenario_md = """
<|{scenario_1}|scenario_dag|>
"""
    Gui(scenario_md).run()
```

**Acceptance Criteria**
- [ ] Ensure new code is unit tested, and check code coverage is at least 90%
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change
- [ ] Ensure any change is well documented

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #787

```
import taipy as tp
from taipy import Config, Orchestrator, Submittable
from taipy.gui import Gui, State

a_cfg = Config.configure_data_node("additional")
...
scenario_cfg = Config.configure_scenario("my_scenario", [task_cfg], [a_cfg])
...
selected_scenario = tp.create_scenario(scenario_cfg)
...
<|{selected_scenario}|scenario_dag|>

```

### CODE REVIEW

File: taipy/gui_core/_adapters.py [hunk 1]
```diff
@@ -131,16 +131,26 @@ def get(self):
...
+                            "type": datanode.entity.storage_type() if hasattr(datanode.entity, "storage_type") else None,
                         }
+                    cat = nodes.get(DataNode.__name__)
+                    if cat is None:
+                        cat = {}
+                        nodes[DataNode.__name__] = cat
+                    for id, datanode in scenario.additional_data_nodes.items():
+                        cat[id] = {
+                            "name": datanode.get_simple_label(),
+                            "type": datanode.storage_type(),
```
Comment (FabienLelaquais) [inline taipy/gui_core/_adapters.py]: No test on `storage_type()` here?

**Review by FabienLelaquais — DISMISSED** [review]
👌

File: taipy/gui_core/_adapters.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui_core/_adapters.py]: every datanode has a storage_type, the other entities don't

**Review by FabienLelaquais — APPROVED** [review]
👌

**Review by dinhlongviolin1 — APPROVED** [review]
LGTM

**Pull request merged**